common/script/index.coffee

```diff
@@ -1168,9 +1168,9 @@ api.wrap = (user, main=true) ->
           user.stats.mp++ if stat is 'int' #increase their MP along with their max MP
         cb? null, _.pick(user,$w 'stats')
 
-      readValentine: (req,cb) ->
-        user.items.special.valentineReceived.shift()
-        user.markModified? 'items.special.valentineReceived'
+      readCard: (cardType, cb) ->
+        user.items.special["#{cardType}Received"].shift()
```

Comment:
This fails on the server:
`"500: Cannot read property &#39;shift&#39; of undefined "`

Maybe we need to put the `cardType` within `req.params` instead of passing it naked?